### PR TITLE
[proposition] code adaptation

### PR DESCRIPTION
**Pull request by ardumont**
Hello,

As you ask for help in milkypostman/melpa#2568, I took the liberty of trying to:
-  so continuing the effort of renaming functions with their respective namespaces. It's the convention already established in org (org-element, ...).

So now every function of org-sync.el starts with `org-sync-`, `org-sync-github.el` starts with `org-sync-github-`, etc...

- fix the documentations (readme, docstring, etc...) accordingly.

- add the tests mechanism i use in different packages i maintain (ardumont/org2jekyll, org-trello/org-trello, ardumont/emacs-celery, etc...). So now, we can add tests in the `test/` folder and trigger them via `make test`. It adds a test dependencies cask/cask. As a next step, we could add tests and travis-ci.

- Adapt according to my understanding of #1 you filed.

# Manual check

As there is no test yet and i cannot in just one evening add tests everywhere, i did a basic manual test before and after the modification.

I created an issues.org before and after the change with org-trello/org-trello url repository and got everything. The result can be seen https://github.com/ardumont/org/blob/master/org-trello/issues.org.

Cheers,

### CODE REVIEW

**Comment by ardumont** [comment]
Hello,

Please, when you'll have time, tell me if something is wrong 
I'll adapt accordingly.

--

The context is:
- i maintain org-trello and i have some very similar use-cases (notably the merge)
- @bzg, the org-mode maintainer told me about org-sync about 2 years ago and i had it in mind ever since... 

So i think I simply could learn a thing or 2 with this and why not help in maintaining this.

--

Also to clarify this `could be badly interpreted` sentence...

> add the tests mechanism `i use in different packages i maintain (ardumont/org2jekyll, org-trello/org-trello, ardumont/emacs-celery, etc...)`.

(It's not show-off)

Cask is quite helpful and simple enough to use once the setup is done.
It permits to run tests outside of your live environment, so it starts fresh.

Cheers,

**Comment by arbox** [comment]
Thank you a lot for this amazing contribution! As said I'm rather new in Emacs package development, so I appreciate your effort and the willingness to help.

Let me review your code! I'll do it soon.

**Comment by ardumont** [comment]
> Thank you a lot for this amazing contribution!

Cool.

>  As said I'm rather new in Emacs package development, 

Like anything new, another environment is rather overwhelming at first :D 

> so I appreciate your effort and the willingness to help.
> Let me review your code! I'll do it soon.

Great!

Cheers,

File: contrib/respect-elisp-conventions.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -x
+
+regexp_os='s/os-\(.*\)/org-sync-\1/g'
+
+for i in org-sync*.el; do
+    sed -e $regexp_os -i $i
+done
```
Comment (arbox) [inline contrib/respect-elisp-conventions.sh]: Do you think we need this script in the repo? It was probably a one time conversion action.

Comment (ardumont) [inline contrib/respect-elisp-conventions.sh]: Yeah, no.

It's just to be transparent on how how i did it.
I did that and check everything worked out ok after that.

Now that you've seen it, off with its head :D

File: org-sync.el [hunk 2]
```diff
@@ -878,50 +879,50 @@ sync again.\n\n")
...
-            ;; replace headlines in local-doc
-            (os-replace-headline-by-buglist headline merged)))))
-
-    (os-add-keyword local-doc "TODO" "OPEN | CLOSED")
+        (org-sync-with-backend url
+                               (let* ((cache (org-sync-get-cache org-sync-base-url))
+                                      (last-fetch (org-sync-get-prop :date-cache cache))
+                                      (local-diff (org-sync-buglist-diff cache local))
+                                      remote remote-diff merged merged-diff)
+
```
Comment (arbox) [inline org-sync.el]: Some inconsistencies in the indentetion?

File: Cask [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+       "org-sync-rtm.el")
+
+(development
+ (depends-on "dash")
+ (depends-on "dash-functional")
+ (depends-on "undercover")
+ (depends-on "ert-runner")
+ (depends-on "ert")
+ (depends-on "ert-expectations")
+ (depends-on "el-mock"))
```
Comment (arbox) [inline Cask]: Having a Cask file is obviously a big plus!

File: org-sync.el (hunk 2 above)
Comment (ardumont) [inline org-sync.el]: Ah, that would be `aggressive-indent` mode which aligned the form.

File: Cask (hunk 3 above)
Comment (ardumont) [inline Cask]: It also permits to make package.
I use it to make the package and send them to marmalade https://marmalade-repo.org/ (another package repository)

**Comment by arbox** [comment]
Ok, thank you so much! We'll try to polish this project further!

**Comment by ardumont** [comment]
Cool.

**Comment by bzg** [comment]
Thanks to both of you for your work on org-sync.  Even if the original author expressed doubts on its usefulness and maintainability, I think it's great to use it as a sandbox for ideas on how to connect org-mode with the outside world of bugtrackers.

**Comment by ardumont** [comment]
> Thanks to both of you for your work on org-sync. 

Thanks!

> Even if the original author expressed doubts on its usefulness and maintainability, I think it's great to use it as a sandbox for ideas on how to connect org-mode with the outside world of bugtrackers.

Indeed.

**Pull request merged**